backend/lcfs/web/api/user/repo.py

```diff
@@ -0,0 +1,241 @@
+from lcfs.web.api.user.schema import UserCreate, UserBase, UserHistory
+from lcfs.web.api.base import (
+    PaginationRequestSchema,
+    lcfs_cache_key_builder,
+    apply_filter_conditions,
+    get_field_for_filter,
+)
+
+logger = getLogger("user_repo")
+
+
+class UserRepository:
+    def __init__(
+        self,
+        session: AsyncSession = Depends(get_async_db_session),
+    ):
+        self.session = session
+
+    def apply_filters(self, pagination, conditions):
+        for filter in pagination.filters:
+            filter_value = filter.filter
+            filter_option = filter.type
+            filter_type = filter.filterType
+
+            if filter.field == "role":
+                field = get_field_for_filter(Role, "name")
+                conditions.append(
+                    Role.name.in_(
+                        [RoleEnum(role.strip()) for role in filter_value.split(",")]
+                    )
+                )
+            elif filter.field == "is_active":
+                filter_value = True if filter_value == "Active" else False
+                filter_option = "true" if filter_value else "false"
+                field = get_field_for_filter(UserProfile, "is_active")
+                conditions.append(
+                    apply_filter_conditions(
+                        field,
+                        filter_value,
+                        filter_option,
+                        filter_type,
+                    )
+                )
+            elif filter.field == "first_name":
+                txt = filter_value.split(" ")
+                field1 = get_field_for_filter(UserProfile, "first_name")
+                conditions.append(
+                    apply_filter_conditions(
+                        field1,
+                        txt[0],
+                        filter_option,
+                        filter_type,
+                    )
+                )
+                field2 = get_field_for_filter(UserProfile, "last_name")
+                conditions.append(
+                    apply_filter_conditions(
+                        field2,
+                        txt[1] if len(txt) > 1 else "",
+                        filter_option,
+                        filter_type,
+                    )
+                )
+            else:
+                field = get_field_for_filter(UserProfile, filter.field)
+                conditions.append(
+                    apply_filter_conditions(
+                        field, filter_value, filter_option, filter_type
+                    )
+                )
+
+    @cache(
+        expire=3600 * 24,
+        key_builder=lcfs_cache_key_builder,
+        namespace="users",
+    )  # Cache for 24 hours, already handled to clear cache if any new users are added or existing users are updated.
+    @repo_handler
+    async def get_users_paginated(
+        self,
+        pagination: PaginationRequestSchema,
+    ) -> List[UserBase]:
```

Comment:
We're using suffix 'Schema' for all pydantic models. Can you plz update all the user pydantic models to use this naming convention.

Reply:
updated.